tools/schematics/testing.ts

```diff
@@ -84,16 +82,16 @@ function beforeExit(): void {
   }
 }
 
+let newVersion: string | undefined;
 function publishLibs(reload = false): void {
-  if (!currentVersion || reload) {
-    currentVersion = semver.parse(
+  if (!newVersion || reload) {
+    newVersion = semver.parse(
       JSON.parse(fs.readFileSync('projects/core/package.json', 'utf-8')).version
-    );
+    )?.version;
   }
-
   // Bump version to publish
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  semver.inc(currentVersion!, 'patch');
+  newVersion = semver.inc(newVersion ?? '', 'patch') ?? '';
```

Comment:
IMHO using `!` (non null assertion) was deliberate in this case. Our intention really isn't falling back to `''`. _We just know_ it won't be undefined. But typescript doesn't know.

Alternative way to fix the typesafety - never let `currentVersion` be undefined:
```ts
function getCurrentVersion(): string {
   const result = semver.parse(
      JSON.parse(fs.readFileSync('projects/core/package.json', 'utf-8')).version
    )?.version;
    if(!result) {
       throw new Error(`File 'projects/core/package.json' doesn't contain a valid field "version"`);
    }
    return result;
}

let currentVersion: string = getCurrentVersion();

function publishLibs(reload = false): void {
  if(reload) {
    currentVersion = getCurrentVersion();
  }
  currentVersion = semver.inc(currentVersion, 'patch');
  ...
}
```

Reply:
Thanks! Done.